serialhub.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2022 Arduino SA
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// Supports Windows, Linux, Mac, BeagleBone Black, and Raspberry Pi
+
+package main
+
+import (
+	"strings"
+	"sync"
+)
+
+type serialhub struct {
+	// Opened serial ports.
+	ports map[*serport]bool
+	mu    sync.Mutex
```

Comment:
I usually see the other way around first the mutex and then the resource that is protected
```suggestion
	mu    sync.Mutex
        ports map[*serport]bool
```

Reply:
I agree but I would leave this code as is for now.